editor/src/components/canvas/dom-walker.ts

```diff
@@ -232,6 +341,27 @@ export function useDomWalker(props: CanvasContainerProps): React.Ref<HTMLDivElem
           if (sceneIndexAttr != null && validPathsAttr != null) {
             const scenePath = TP.fromString(sceneIndexAttr.value)
             const validPaths = validPathsAttr.value.split(' ')
+            const sceneID = sceneIndexAttr.value
+            if (
+              !ObserversAvailable ||
+              invalidatedSceneIDsRef.current == null ||
+              invalidatedSceneIDsRef.current.indexOf(sceneID) > -1
+            ) {
+              invalidatedSceneIDsRef.current = invalidatedSceneIDsRef.current.filter(
+                (sceneIDref) => sceneIDref !== sceneID,
+              )
+            } else {
+              const elementFromCurrentMetadata = MetadataUtils.findElementMetadata(
+                scenePath,
+                rootMetadataInStateRef.current,
+              )
+              if (elementFromCurrentMetadata != null) {
+                // early return for cached scenes
+                rootMetadata.push(...elementFromCurrentMetadata.children)
+                rootMetadata.push(elementFromCurrentMetadata)
+                return
```

Comment:
Please can you restructure this code to avoid having this early return in the middle of the function

Reply:
fixed